src/main/java/TaskHandler.java

```diff
@@ -0,0 +1,186 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TaskHandler{
+
+    public static void addTasks(String input, ArrayList<Task> storage, Path filePath) throws IOException {
```

Comment:
You can try to separate the "adding task" component and "command" components. eg. "command" components include marking and unmarking tasks. This is because this addTasks method is quite long.